web/src/stream_events.js

```diff
@@ -209,6 +213,6 @@ export function process_subscriber_update(user_ids, stream_ids) {
         const sub = sub_store.get(stream_id);
         stream_settings_ui.update_subscribers_ui(sub);
     }
-    compose_fade.update_faded_users();
     user_profile.update_user_profile_streams_list_for_users(user_ids);
+    activity_ui.build_user_sidebar();
```

Comment:
Can this logic be conditional on a stream being modified being the one you're looking at? This rebuild is a somewhat expensive operation, and in a big organization, the `peer_add` events (which are decently common) will usually be related to a stream that isn't your current view (because DMs, inbox, recent, all messages never match, and there's probably a lot of streams, most of which are irrelevant).

Reply:
Also please move this commit to the end -- and I think you can squash all the other older fixup commits except this and the `compose_fade` one (which is a nice atomic feature removal), following rebasing on top of https://github.com/zulip/zulip/pull/28848.

Reply:
To clarify -- you want the PR to be [every commit except this commit and the compose_fade one, squashed together] and then [the compose_fade commit] and then [this commit] so three total? Or do you want more separation than that still?

Reply:
> Can this logic be conditional on a stream being modified being the one you're looking at? 

Yup, updated the commit! Also moved it to the end